Versioner/UpdateInfo.swift

```diff
@@ -9,82 +9,47 @@
 import Foundation
 import UIKit
 
-public enum UpdateInfoError: Error {
-    case invalidJsonData
-    case invalidLatestVersion
-    case invalidCurrentVersion
-    case unknown(String?)
+protocol UpdateInfoValues {
+    var minimumRequiredVersion: Version? { get }
+    var minimumSdkForMinimumRequiredVersion: Version? { get }
+    var latestVersion: Version { get }
+    var minimumSdkForLatestVersion: Version? { get }
+    var installedVersion: Version { get }
+    var sdkVersion: Version { get }
+    var isMinimumVersionSatisfied: Bool { get }
+    var metadata: [String: Any]? { get }
 }
 
-public struct UpdateInfo {
+public struct UpdateInfo{
 
     public enum NotificationType : String {
         case always = "ALWAYS"
         case once = "ONCE"
     }
 
-    /**
-     Returns minimum required version of the app.
-     */
-    public private(set) var minimumRequiredVersion: Version?
-    
-    /**
-     Returns minimum sdk for minimum required version of the app.
-     */
-    public private(set) var minimumSdkForMinimumRequiredVersion: Version?
+    // MARK: - Private properties
+    private var minRequiredVersion: Version?
+    private var minSdkForMinimumRequiredVersion: Version?
+    private var latestVers: Version
```

Comment:
What is with this naming? Even if you don't use `_` for privates, this excellent place where they should be used instead of this weird naming when you cut off half of the string